src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -66,3 +67,23 @@ const getNextPage = (beneficialOwnerTypeChoices: BeneficialOwnerTypeChoice | Man
   }
 };
 
+const retrieveBeneficialOwners = async (req: Request, appData: ApplicationData) => {
+  const session = req.session as Session;
+  const pscs: CompanyPersonsWithSignificantControl = await getCompanyPsc(req, appData[EntityNumberKey] as string);
+  if (pscs !== undefined) {
+    for (const psc of (pscs.items || [])) {
+      if (psc.kind === "individual-person-with-significant-control"){
+        const beneficialOwnerI: BeneficialOwnerIndividual = mapPscToBeneficialOwnerTypeIndividual(psc);
+        setApplicationData(session, beneficialOwnerI, BeneficialOwnerIndividualKey);
+      } else if (psc.kind === "corporate-entity-beneficial-owner") {
+        const beneficialOwnerOther: BeneficialOwnerOther = mapPscToBeneficialOwnerOther(psc);
+        setApplicationData(session, beneficialOwnerOther, BeneficialOwnerOtherKey);
+      } else if (psc.kind === "legal-person-with-significant-control") {
+        const beneficialOwnerGov: BeneficialOwnerGov = mapPscToBeneficialOwnerGov(psc);
+        setApplicationData(session, beneficialOwnerGov, BeneficialOwnerGovKey);
+      }
+    }
+  } else {
+    setFetchedBoMoData(appData);
```

Comment:
Would it be better to split this into
      setFetchedBoData
      setFetchedMoData
Similarly for the hasFetched ?
This would more cleanly decouple BOs and MOs ?

Reply:
Both hasFetchedBoAndMoData and setFetchedBoAndMoData only sets bo_mo_data to true. 

Separating it doesn't add any benefit I think, if false, call both api's, if not, don't.

There is another ticket for the MO

Reply:
MOs will be retrieved after BOs and then the flag set when we add: https://github.com/companieshouse/overseas-entities-web/pull/749